test/cypress/integration/widgets/blueprints_spec.ts

```diff
@@ -514,4 +514,32 @@ describe('Blueprints widget', () => {
             cy.get('.modal .tabular > a.item').should('have.length', marketplaceTabs.length - 1);
         });
     });
+
+    describe('for blueprint with required secrets', () => {
+        const secretName = 'test';
+        const blueprintName = 'blueprint_with_required_secrets';
+        before(() => cy.uploadBlueprint('blueprints/required_secrets.zip', blueprintName).refreshPage());
+        after(() => cy.stageRequest(`/console/sp/secrets/${secretName}`, 'DELETE'));
+
+        it('should open Missing Secrets Error message when deploying blueprint with missing secrets', () => {
+            getBlueprintRow(blueprintName).find('.rocket').click();
+            cy.get('input[name=deploymentName]').type('blahBlahBlah');
+            cy.contains('.modal button', 'Deploy').click();
+            cy.get('form.error .error .header').should('have.text', 'Missing Secrets Error');
+            cy.get('form.error .error p').should(
+                'have.text',
+                'The following required secrets are missing in this tenant:'
+            );
+            cy.get('form.error .error .item').should('have.text', secretName);
```

Comment:
You can probably use `.within` to make it a bit more structured:
```suggestion
            cy.get('form.error .error').within(() => {
              cy.get('.header').should('have.text', 'Missing Secrets Error');
              cy.get('p').should('have.text', 'The following required secrets are missing in this tenant:');
              cy.get('.item').should('have.text', secretName);
            });
```

Reply:
I changed to using `.within`